odl/discr/lp_discr.py

```diff
@@ -104,19 +94,61 @@ def __init__(self, fspace, partition, tspace, **kwargs):
             raise ValueError('got unexpected keyword arguments {}'
                              ''.format(kwargs))
 
+    # --- Meta-info
+
     @property
-    def axis_labels(self):
-        """Labels for axes when displaying space elements."""
-        return self.__axis_labels
+    def element_type(self):
+        """`DiscretizedSpaceElement`"""
+        return DiscretizedSpaceElement
+
+    # --- Constructor args
 
     @property
     def partition(self):
         """`RectPartition` of the function domain."""
         return self.__partition
 
+    @property
+    def tspace(self):
+        """Space for the coefficients of the elements of this space."""
+        return self.__tspace
+
+    @property
+    def axis_labels(self):
+        """Labels for axes when displaying space elements."""
+        return self.__axis_labels
+
+    # --- Pass-through `partition` attributes
+
+    @property
+    def domain(self):
+        """Set on which functions are defined before discretization."""
+        return self.partition.set
+
+    # --- Pass-through `tspace` attributes
+
+    @property
+    def weighting(self):
+        """This space's weighting scheme."""
+        # TODO(kohr-h): `weighting` is optional in `tspace`, how should we
+        # handle that?
+        return self.tspace.weighting
+
+    @property
+    def is_weighted(self):
+        """``True`` if the ``tspace`` is weighted."""
+        return getattr(self.tspace, 'is_weighted', False)
+
+    @property
+    def impl(self):
+        """Name of the implementation back-end."""
+        return self.tspace.impl
+
     @property
     def exponent(self):
         """Exponent of this space, the ``p`` in ``L^p``."""
+        # TODO(kohr-h): `exponent` is optional in `tspace`, how should we
```

Comment:
Isn't this important? What happens if the underlying tensor space has no exponent, but I want my discretized space to have one? Or is the philosophy (at least at this moment) to simply leave it to the user to make sure that things are consistent?

Reply:
Yes, it is important. But the paradigm here is that the exponent is simply whatever the exponent of the wrapped `tspace` is. If it doesn't exist, then the `AttributeError` will just bubble up. I'm not sure how we can do better than that.